pgjdbc/src/main/java/org/postgresql/jdbc/PgDatabaseMetaData.java

```diff
@@ -2541,7 +2542,43 @@ public boolean isWrapperFor(Class<?> iface) throws SQLException {
 
   public ResultSet getFunctions(String catalog, String schemaPattern, String functionNamePattern)
       throws SQLException {
-    return getProcedures(catalog, schemaPattern, functionNamePattern);
+
+    // The pg_get_function_result only exists 8.4 or later
+    String ver = connection.getDBVersionNumber();
+    Version dbVer = ServerVersion.from(ver);
+    boolean pgFuncResultExists = dbVer.getVersionNum() >= ServerVersion.v8_4.getVersionNum();
```

Comment:
I think the proper way to check server version support is using `connection.haveMinimumServerVersion(ServerVersion.v8_4)`.

Reply:
@jorsol, Thanks for the good suggestion. I pushed an update for this now.